### PR TITLE
UNDERTOW:

### PR DESCRIPTION
**Pull request by manolama**
- Add Stijn de Witt's cors handler to Undertow and use almost the old settings
  from TSDB 2.x

### CODE REVIEW

File: implementation/server-undertow/src/main/java/net/opentsdb/tsd/TSDMain.java [hunk 1]
```diff
@@ -229,15 +243,24 @@ public static void main(final String[] args) {
...
+      if (!Strings.isNullOrEmpty(tsdb.getConfig().getString(CORS_PATTERN_KEY))) {
+        handler = new com.stijndewitt.undertow.cors.Filter(handler);
+        ((com.stijndewitt.undertow.cors.Filter) handler).setUrlPattern(
+            tsdb.getConfig().getString(CORS_PATTERN_KEY));
+        ((com.stijndewitt.undertow.cors.Filter) handler).setAllowHeaders(
+            tsdb.getConfig().getString(CORS_HEADERS_KEY));
+      }
+      
+      // support compression!
+      handler = new EncodingHandler.Builder()
```
Comment (gskchaitanya) [inline implementation/server-undertow/src/main/java/net/opentsdb/tsd/TSDMain.java]: Might be a good idea to expose the Undertow IO/workertask threads as a config

Comment (manolama) [reply implementation/server-undertow/src/main/java/net/opentsdb/tsd/TSDMain.java]: yeah I'll get a ton of configs in a future pr.

**Comment by manolama** [comment]
Merged in 0487d76dfeb10267a9137f033d30e65d6bc8a31b

File: implementation/server-undertow/src/main/java/net/opentsdb/tsd/TSDMain.java (hunk 1 above)
Comment (gskchaitanya) [reply implementation/server-undertow/src/main/java/net/opentsdb/tsd/TSDMain.java]: Right on!

**Pull request closed without merging**